homework_03/05_Viktoriya_Vasileva/Circles.py

```diff
@@ -0,0 +1,69 @@
+import enum
+import math
+
+class RelativePosition(enum.Enum):
+    NO_COMMON_POINTS = 0
+    TOUCHING = 1
+    INTERSECTING = 2
+    #TOUCHING_INTERNALLY = 3
+    SAME = 3
+
+class Point:
+    def __init__(self, x, y):
+        assert isinstance(x, float) 
+        assert isinstance(y, float)
+        self.x = x
+        self.y = y
+
+    def line(self, point):
+        assert isinstance(point, Point)
+        result = round(math.sqrt( math.pow(point.x - self.x, 2) + math.pow(point.y - self.y, 2)), 5)
+        return result
+        
+class Circle:
+    def __init__(self, point, r):
+        assert isinstance(point, Point)
+        assert isinstance(r, float)
+        assert round(r, 5) > 0
+        self.point = point
+        self.r = r
+
+    def find_relative_position(self, circle):
+        assert isinstance(circle, Circle)
+
+       #common center
+        if self.point.x == circle.point.x and self.point.y == circle.point.y:
+            return RelativePosition.SAME
+
+        AB = self.point.line(circle.point) # >=0
+        radius = round(self.r + circle.r, 5)
+        #no common points
+        if AB > radius:
+            return RelativePosition.NO_COMMON_POINTS
+        #touching
+        if AB == radius:
+            return RelativePosition.TOUCHING
+        
+        radius_sub = self.r - circle.r
```

Comment:
Here you can use directly abs(self.r - circle.r). This way you will skip the next check on lines 48 and 49.

Reply:
As Teodor it is better to call abs() so you can avoid using it below!